Yosemite/Yosemite/Tools/POS/PointOfSaleOrderService.swift

```diff
@@ -0,0 +1,219 @@
+
+    public init(itemID: Int64?, product: PointOfSaleCartProduct, quantity: Decimal) {
+        self.itemID = itemID
+        self.product = product
+        self.quantity = quantity
+    }
+}
+
+public struct PointOfSaleOrder {
+    public let siteID: Int64
+    public let orderID: Int64
+    public let total: String
+    public let totalTax: String
+    public let currency: String
+    let items: [PointOfSaleOrderItem]
+
+    static func createFromOrder(_ order: Order) -> PointOfSaleOrder {
+        return PointOfSaleOrder(siteID: order.siteID,
+                                orderID: order.orderID,
+                                total: order.total,
+                                totalTax: order.totalTax,
+                                currency: order.currency,
+                                items: order.items.map { PointOfSaleOrderItem(orderItem: $0) })
+    }
+}
+
+struct PointOfSaleOrderItem {
+    let itemID: Int64
+
+    /// The product ID of a product order item, or the ID of the variable product if the order item is a product variation.
+    let productID: Int64
+    let quantity: Decimal
+
+    func toOrderItem() -> OrderItem {
+        .init(itemID: itemID,
+              name: "",
+              productID: productID,
+              variationID: .zero,
+              quantity: quantity,
+              price: .zero,
+              sku: nil,
+              subtotal: "",
+              subtotalTax: "",
+              taxClass: "",
+              taxes: [],
+              total: "",
+              totalTax: "",
+              attributes: [],
+              addOns: [],
+              parent: nil,
+              bundleConfiguration: [])
+    }
+
+    init(orderItem: OrderItem) {
+        self.itemID = orderItem.itemID
+        self.productID = orderItem.productID
+        self.quantity = orderItem.quantity
+    }
+}
+
+public protocol PointOfSaleOrderServiceProtocol {
+    /// Syncs order based on the cart.
+    /// - Parameters:
+    ///   - cart: Cart with optional items (product & quantity).
+    ///   - order: Optional latest remotely synced order. Nil when syncing order for the first time.
+    ///   - allProducts: Necessary for removing existing order items with products that have been removed from the cart.
+    /// - Returns: Order from the remote sync.
+    func syncOrder(cart: [PointOfSaleCartItem], order: PointOfSaleOrder?, allProducts: [PointOfSaleCartProduct]) async throws -> PointOfSaleOrder
+
+    /// Updates status of an order and syncs it
+    /// - Parameters:
+    ///   - posOrder: POS order.
+    ///   - status: New order status.
+    /// - Returns: Updated and synced POS order.
+    func updateOrderStatus(posOrder: PointOfSaleOrder, status: OrderStatusEnum) async throws -> PointOfSaleOrder
+
+    /// Creates WOO Order from POS Order.
+    /// - Parameters:
+    ///   - posOrder: Optional POS order.
+    /// - Returns: Order created from posOrder data.
+    func orderFrom(posOrder: PointOfSaleOrder?) -> Order
```

Comment:
nit: feels more swifty

```suggestion
    func order(from posOrder: PointOfSaleOrder?) -> Order
```

Reply:
I tried checking the use case for this function in the previous PR https://github.com/woocommerce/woocommerce-ios/pull/13029, it feels a bit odd how a nil POS order could return a non-nil Order. Since the use case is meant for taking payment with an Order given the current payment service API and we only take payment for a non-nil order, how about making the `posOrder: PointOfSaleOrder` parameter non-optional?

```suggestion
    func orderFrom(posOrder: PointOfSaleOrder) -> Order
```

Reply:
I made it optional since the POS order that we used could be nil and then we either just create empty one or create it by copying. But I can change it not to be nil and later update our model to have a non nil empty POS order 👍